src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,505 @@
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            public ImmutablePair<BreakpointEvidence, Boolean> next() {
+                if ( !hasNext() ) {
+                    throw new NoSuchElementException("No next element.");
+                }
+                final BreakpointEvidence overlapper = listItr.next();
+                Boolean isCoherent = false;
+                if(checkCoherence) {
+                    Boolean overlapperStrand = overlapper.isEvidenceUpstreamOfBreakpoint();
+                    if(overlapperStrand == isForwardStrand && overlapper.hasDistalTargets(readMetadata, minEvidenceMapQ)) {
+                        for(final StrandedInterval distalTarget : distalTargets) {
+                            for(final StrandedInterval overlapperDistalTarget : overlapper.getDistalTargets(readMetadata, minEvidenceMapQ)) {
+                                if(distalTarget.getStrand() == overlapperDistalTarget.getStrand()
+                                        && distalTarget.getInterval().overlaps(overlapperDistalTarget.getInterval())) {
+                                    isCoherent = true;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+                return new ImmutablePair<>(overlapper, isCoherent);
+            }
+        }
+    }
+    */
+
+    private static class UnscaledOverlapInfo {
+        final int numOverlap;
+        final int numCoherent;
+        final int overlapMappingQuality;
+        final int coherentMappingQuality;
+        UnscaledOverlapInfo(final int numOverlap, final int numCoherent, final int overlapMappingQuality,
+                            final int coherentMappingQuality) {
+            this.numOverlap = numOverlap;
+            this.numCoherent = numCoherent;
+            this.overlapMappingQuality = overlapMappingQuality;
+            this.coherentMappingQuality = coherentMappingQuality;
+        }
+    }
+
+    /**
+     * Class that takes raw overlap info and automatically scales it to meanGenomeCoverage, storing the result for later retrieval
+     */
+    private static class CoverageScaledOverlapInfo {
+        final double numOverlap;
+        final double overlapMappingQuality;
+        final double meanOverlapMappingQuality;
+        final double numCoherent;
+        final double coherentMappingQuality;
+
+        CoverageScaledOverlapInfo(final int numOverlap, final int numCoherent, final int overlapMappingQuality,
+                                  final int coherentMappingQuality, final double meanOverlapMappingQuality,
+                                  final double coverage) {
+            this.numOverlap = ((double)numOverlap) / coverage;
+            this.overlapMappingQuality = ((double)overlapMappingQuality) / coverage;
```

Comment:
Now that I look at this again I'm a bit confused about dividing mapping quality by mean coverage, here and for coherentMappingQuality. That seems somewhat strange to me. What is the reasoning for doing that?

Reply:
totalOverlapMappingQuality produces a quantity very similar to numOverlappers, but scaled by mapping quality (e.g. high for overlappers with good mapping quality). Similar to numOverlappers, it needs to be scaled by coverage to ensure proper evaluation of BAMS that have coverage higher or lower than the training data set.